defaults/main.yml

```diff
@@ -20,7 +20,7 @@ os_images_package_state: present
 os_images_upper_constraints_file: "https://releases.openstack.org/constraints/upper/{% if ansible_facts.python.version.major == 2 %}train{% else %}yoga{% endif %}"
 
 # Upper constraints file for installation of DIB to build images.
-os_images_dib_upper_constraints_file: "{{ os_images_upper_constraints_file }}"
+os_images_dib_upper_constraints_file: ""
```

Comment:
Seems like this could lead to instability, if we install without upper constraints. It also breaks python 2, in case we still care about that (if not we should drop the code above).

Would it be better to override this variable in the calling code, until a suitable version of DIB is available in UC?

Reply:
DIB won't get updated to 3.23.1 in master (Zed) UC - unless somebody raises a manual patch - and Ian mentioned yesterday he's not planning this.
I think it's second occurrence of this (last time happened in Yoga and I had to propose a DIB raise in UC), and I doubt we need UC for DIB (since it's branchless and doesn't use UC in it's own CI jobs).
I agree on the python2 step, but maybe we should just notify users we're not supporting python2 anymore?
Python 3.5 is EOL now, how long do we need to drag the weight of python2?
If we still need it - we can just do an if python2 - then use Train UC, if no - don't use any.

The alternative is that we do what I did in Yoga and keep using UC - should we vote? ;-)

Reply:
Ok, let's see how we go without. DIB may be branchless, but we may run into weird issues if we start installing the latest DIB on old OpenStack releases.

Reply:
We can always revisit this or fix it locally on old OpenStack releases.

Reply:
With the separation of upper constraints in #67, we have much less risk of running into an issue. DIB itself has few dependencies, it would only cause breakage if it bumped a library also used by openstacksdk or OSC, such as stevedore, to a version that was incompatible.